internal/services/dispatcher/dispatcher_v1.go

```diff
@@ -226,12 +226,42 @@ func (d *DispatcherImpl) handleTaskBulkAssignedTask(ctx context.Context, msg *ms
 			continue
 		}
 
+		retrivePayloadOpts := make([]v1.RetrievePayloadOpts, len(bulkDatas))
+
+		for i, task := range bulkDatas {
+			retrivePayloadOpts[i] = v1.RetrievePayloadOpts{
+				Id:         task.ID,
+				InsertedAt: task.InsertedAt,
+				Type:       sqlcv1.V1PayloadTypeTASKINPUT,
+				TenantId:   task.TenantID,
+			}
+		}
+
+		inputs, err := d.repov1.Payloads().BulkRetrieve(ctx, retrivePayloadOpts...)
+
+		if err != nil {
+			d.l.Error().Err(err).Msgf("could not bulk retrieve inputs for %d tasks", len(bulkDatas))
+		}
+
 		for _, task := range bulkDatas {
+			input, ok := inputs[v1.RetrievePayloadOpts{
+				Id:         task.ID,
+				InsertedAt: task.InsertedAt,
+				Type:       sqlcv1.V1PayloadTypeTASKINPUT,
+				TenantId:   task.TenantID,
+			}]
+
+			if input == nil || !ok {
+				// If the input wasn't found in the payload store,
+				// fall back to the input stored on the task itself.
+				input = task.Input
+			}
```

Comment:
doesn't the payload store repository use this fallback?

Reply:
not sure I follow the question here - do you mean this is redundant?

Reply:
yes, isn't calling `BulkRetrieve` on the payload store safe to do without needing to fall back each time?

Reply:
I don't think so? this is for handling the case where e.g. a DAG started before the upgrade and then the second task runs afterwards, in which case I think there might be no data in the payloads table yet. let me know if you're thinking of something different here!